### ISSUE TITLE
Default error type does not match implementation

### ISSUE THREAD

**Issue by Fryuni** [Sytten/nexus-shield#147]
The typing for the `defaultError` parameter is:
```
declare type ErrorMapper = (err: Error, parent: any, args: any, ctx: ShieldContext, info: GraphQLResolveInfo) => Promise<Error> | Error;
declare type DefaultError = Error | ErrorMapper;
export interface ShieldPluginSettings {
    defaultError?: DefaultError;
}
```
It should accept a function that creates the final error.

But the implementation [here](https://github.com/Sytten/nexus-shield/blob/1ad86f105ac0c7572d79ee836fabb14ad6ee996c/src/plugin.ts#L95) is to just throw the default error, which wont work if it is a function.

**Comment by Sytten** [comment]
Good catch I will fix that ASAP!

**Comment by Sytten** [comment]
Fixed in v1.1.2, I did change the Mapper type since I dont catch error in this plugin I cant forward it to the function.
